lib/Parse/ParseStmt.cpp

```diff
@@ -1833,6 +1833,74 @@ Parser::parseStmtConditionElement(SmallVectorImpl<StmtConditionElement> &result,
   return Status;
 }
 
+/// Returns `true` if the current token represents the start of a conditional
+/// statement body.
+bool Parser::isStartOfConditionalStmtBody() {
+  Parser::BacktrackingScope Backtrack(*this);
+
+  if (!Tok.is(tok::l_brace)) {
+    // Statement bodies always start with a '{'. If there is no '{', we can't be
+    // at the statement body.
+    return false;
+  }
+
+  skipSingle();
```

Comment:
Does `BacktrackingScope` need to be created in case we return early here since there was not `consume` yet. Since is a very often case, we could avoid creating it and save of a few unnecessary operations if that makes sense.
```suggestion
  if (!Tok.is(tok::l_brace)) {
    // Statement bodies always start with a '{'. If there is no '{', we can't be
    // at the statement body.
    return false;
  }
  
  Parser::BacktrackingScope Backtrack(*this);
  
  skipSingle();
```